clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -1432,4 +1425,21 @@ public static String[] enumOptions(Class<? extends Enum<?>> enumClass) {
                 .toArray(String[]::new);
     }
 
+    /**
+     * Creates a preallocated file
```

Comment:
Please add a note that the responsibility to free up resource i.e. close the FileChannel is transferred to the caller of this function.